gobblin-api/src/main/java/org/apache/gobblin/source/extractor/ComparableWatermark.java

```diff
@@ -20,7 +20,7 @@
 /**
  * {@link Watermark} that is also {@link Comparable}.
  */
-public interface ComparableWatermark<V extends Comparable<V>> extends Watermark, Comparable<ComparableWatermark>{
+public interface ComparableWatermark<V> extends Watermark, Comparable<ComparableWatermark>{
```

Comment:
Would this break other existing use case that have already implements the interface ?
or Is creating a new interface have a clean start ?

Reply:
This change enlarge the scope of the general class. And we have two implementation of this calls, LongWatermark and StringWatermark, both are not affected with this change

Reply:
Fro other classes that use ComparableWatermark, as long as the watermark itself is still comparable, we should be fine